tests/trinity/core/integration_test_helpers.py

```diff
@@ -96,3 +106,52 @@ class FakeAsyncChain(MiningChain):
     coro_import_block = coro_import_block
     coro_validate_chain = async_passthrough('validate_chain')
     coro_validate_receipt = async_passthrough('validate_receipt')
+
+
+class ByzantiumTestChain(FakeAsyncChain):
+    vm_configuration = ((0, ByzantiumVM),)
+    chaindb_class = FakeAsyncChainDB
+    network_id = 999
+
+
+FUNDED_ACCT = keys.PrivateKey(
+    decode_hex("49a7b37aa6f6645917e7b807e9d1c00d4fa71f18343b0d4122a4d2df64dd6fee"))
+
+
+def load_mining_chain(db):
+    class POWByzantiumVM(POWMiningMixin, ByzantiumVM):
+        pass
+
+    class PoWMiningChain(ByzantiumTestChain):
+        vm_configuration = ((0, POWByzantiumVM),)
+
+    GENESIS_PARAMS = {
+        'parent_hash': constants.GENESIS_PARENT_HASH,
+        'uncles_hash': constants.EMPTY_UNCLE_HASH,
+        'coinbase': constants.ZERO_ADDRESS,
+        'transaction_root': constants.BLANK_ROOT_HASH,
+        'receipt_root': constants.BLANK_ROOT_HASH,
+        'bloom': 0,
+        'difficulty': 5,
+        'block_number': constants.GENESIS_BLOCK_NUMBER,
+        'gas_limit': 3141592,
+        'gas_used': 0,
+        'timestamp': 1514764800,
+        'extra_data': constants.GENESIS_EXTRA_DATA,
+        'nonce': constants.GENESIS_NONCE
+    }
+
+    GENESIS_STATE = {
+        FUNDED_ACCT.public_key.to_canonical_address(): {
+            "balance": 100000000000000000,
+            "code": b"",
+            "nonce": 0,
+            "storage": {}
+        }
+    }
+
+    return PoWMiningChain.from_genesis(db, GENESIS_PARAMS, GENESIS_STATE)
```

Comment:
This whole function seems like it should use the chain builder API.  Any reason you didn't?  I suspect it'd be significantly shorter and still easy to read and see what it is doing.

Reply:
It was a fairly dumb copy from the old implementation. I'll try out the chain builder API.